source/util.cpp

```diff
@@ -73,6 +73,23 @@ string replace_(string const &s, string const & from, string const &to)
 	return r;
 }
 
+/// Move all occurances of non-alphanumerical characters to the end of the string and replace them with their ASCII codes.
+std::string to_alphanumerical(const std::string& input)
+{
+    std::string output;
+    std::string suffix;
+    size_t pos=0;
+    for (auto c: input) {
+    
+        if ( ( c>='0' && c<='9') ||  ( c>='a' && c<='z') || ( c>='A' && c<='Z') ||  c=='_' )
+            output+=c;
+        else
+            { suffix+='_'; suffix+=std::to_string(pos); suffix+='_'; suffix+=std::to_string(int(c)); }
+     pos++;
+     }
+    return suffix.size()?output+"fixed"+suffix:output;
+}
+
```

Comment:
why i am not directly oppose moving suffixes to the end but this strategy may seem a bit confusing to user, - why not _replace_ characters with ascii codes instead? Moreover, i am guessing that you want to add ASCII codes to make function names unique? If so then this is not fully "bullet proof" strategy since it is possible that two includes exist: one with special characters and another with numeric suffixes. To address this better would be to convert to alphanumeric at name-prefix level (ie line ~376 of context.cpp) and allow current code to handle duplicates for us. This will also lead to much more readable file names.